document/core/exec/instructions.rst

```diff
@@ -388,6 +399,93 @@ Memory Instructions
    However, it may be substantially slower on some hardware.
 
 
+.. _exec-memory.size:
```

Comment:
I would strongly recommend not moving unrelated stuff around for now, since that will cause major headaches with continuous merging of changes from upstream (especially with the memory ops). It also makes reviewing more difficult.

If you want to move stuff for clean-up, maybe insert a `todo::` as a reminder to do that once the proposal is closer to merging.